include/picongpu/particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp

```diff
@@ -63,40 +63,27 @@ namespace ionization
          * @tparam ParticleType type of particle for which to calculate
          *     an average charge state
          *
-         * @param density number density value
-         * @param kinEnergyDensity kinetic energy density value
+         * @param temperature electron "temperature" value calculated from average
+         *        kinetic electron energy per ion in units of eV
+         * @param massDensity ion mass density in units of g/cm^3
          *
          * @return average charge state prediction according to the Thomas-Fermi model
          */
         template< typename ParticleType >
         HDINLINE float_X
-        detailedBalanceThomasFermi( float_X const kinEnergyDensity, float_X const density, ParticleType & parentIon )
+        detailedBalanceThomasFermi( float_X const temperature, float_X const massDensity, ParticleType & parentIon )
         {
 
             /* @TODO replace the float_64 with float_X and make sure the values are scaled to PIConGPU units */
-            constexpr float_64 protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
-            constexpr float_64 neutronNumber = GetAtomicNumbers<ParticleType>::type::numberOfNeutrons;
+            constexpr float_64 protonNumber = GetAtomicNumbers< ParticleType >::type::numberOfProtons;
+            constexpr float_64 neutronNumber = GetAtomicNumbers< ParticleType >::type::numberOfNeutrons;
 
             /* atomic mass number (usually A) A = N + Z */
             constexpr float_64 massNumber = neutronNumber + protonNumber;
 
-            /** @TODO replace the static_cast<float_64> by casts to float_X
-             * or leave out entirely and compute everything in PIConGPU scaled units
-             */
-            float_64 const densityUnit = static_cast<float_64>(particleToGrid::derivedAttributes::Density().getUnit()[0]);
-            float_64 const kinEnergyDensityUnit = static_cast<float_64>(particleToGrid::derivedAttributes::EnergyDensity().getUnit()[0]);
-            /* convert from kinetic energy density to average kinetic energy per particle */
-            float_64 const kinEnergyUnit = kinEnergyDensityUnit / densityUnit;
-            float_64 const kinEnergy = kinEnergyDensity / density * kinEnergyUnit;
-            /** convert kinetic energy in J to "temperature" in eV by assuming an ideal electron gas
-             * E_kin = 3/2 k*T
-             */
-            constexpr float_64 convKinEnergyToTemperature = UNITCONV_Joule_to_keV * float_64(1.e3) * float_64(2./3.);
-            float_64 const temperature = kinEnergy * convKinEnergyToTemperature;
-
-            float_64 const T_0 = temperature/math::pow(protonNumber,float_64(4./3.));
+            float_64 const T_0 = temperature/math::pow( protonNumber,float_64( 4./3. ) );
```

Comment:
just when you are on it@ missing space after `,` and around operator `/`

Reply:
one day I'll get them all...

Reply:
hopefully, `clang-format` will get them all for us :) (with llvm 5.0+)